### PR TITLE
More rename from GPU thread to raster thread

### PR DESCRIPTION
**Pull request by liyuqian**
This PR touches variable names, class names, and file names so it's significantly more risky than its predecessor https://github.com/flutter/engine/pull/17329

Due to file name changes, this PR is expected to change the license files.

We haven't rename `shell/gpu` to `shell/raster` yet. It should be optional but I think it's better to have `raster_surface_software.cc` than `gpu_surface_software.cc`.

### CODE REVIEW

File: shell/common/shell.cc [hunk 1]
```diff
@@ -626,15 +626,15 @@ void Shell::OnPlatformViewCreated(std::unique_ptr<Surface> surface) {
...
-  bool should_post_gpu_task =
-      task_runners_.GetGPUTaskRunner() != task_runners_.GetPlatformTaskRunner();
-
-  auto ui_task = [engine = engine_->GetWeakPtr(),                      //
-                  gpu_task_runner = task_runners_.GetGPUTaskRunner(),  //
+  // latch. To avoid the described deadlock, if the raster and the platform
+  // threads are the same, should_post_gpu_task will be false, and then instead
+  // of posting a task to the raster thread, the ui thread just signals the
+  // latch and the platform/raster thread follows with executing gpu_task.
+  bool should_post_gpu_task = task_runners_.GetRasterTaskRunner() !=
```
Comment (chinmaygarde) [inline shell/common/shell.cc]: Missing var update.

File: shell/common/shell.cc [hunk 2]
```diff
@@ -721,24 +721,24 @@ void Shell::OnPlatformViewDestroyed() {
   // gpu_task to the raster thread triggers signaling the latch(on the IO
   // thread). If the GPU the and platform threads are the same this results in a
```
Comment (chinmaygarde) [inline shell/common/shell.cc]: This comment uses the old names still.

File: fml/raster_thread_merger.h [hunk 3]
```diff
@@ -27,14 +32,14 @@ class GpuThreadMerger : public fml::RefCountedThreadSafe<GpuThreadMerger> {
 
   void ExtendLeaseTo(size_t lease_term);
 
-  // Returns |GpuThreadStatus::kUnmergedNow| if this call resulted in splitting
-  // the GPU and platform threads. Reduces the lease term by 1.
-  GpuThreadStatus DecrementLease();
+  // Returns |RasterThreadStatus::kUnmergedNow| if this call resulted in
+  // splitting the GPU and platform threads. Reduces the lease term by 1.
```
Comment (chinmaygarde) [inline fml/raster_thread_merger.h]: Outdated thread name in comment.

File: lib/ui/compositing/scene_host.cc [hunk 4]
```diff
@@ -190,7 +190,7 @@ SceneHost::SceneHost(fml::RefPtr<zircon::dart::Handle> viewHolderToken,
 
   // Pass the raw handle to the GPU thead; destroying a |zircon::dart::Handle|
```
Comment (chinmaygarde) [inline lib/ui/compositing/scene_host.cc]: Outdated thread name in comment.

File: lib/ui/painting/picture.cc [hunk 5]
```diff
@@ -117,7 +117,7 @@ Dart_Handle Picture::RasterizeToImage(sk_sp<SkPicture> picture,
 
   // Kick things off on the GPU.
```
Comment (chinmaygarde) [inline lib/ui/painting/picture.cc]: "on the raster task runner"

File: shell/common/shell.cc [hunk 6]
```diff
@@ -721,24 +721,24 @@ void Shell::OnPlatformViewDestroyed() {
...
   // deadlock as the gpu_task will never be posted to the plaform/raster thread
-  // that is blocked on a latch.  To avoid the described deadlock, if the gpu
+  // that is blocked on a latch.  To avoid the described deadlock, if the raster
   // and the platform threads are the same, should_post_gpu_task will be false,
   // and then instead of posting a task to the raster thread, the ui thread just
   // signals the latch and the platform/raster thread follows with executing
   // gpu_task.
```
Comment (chinmaygarde) [inline shell/common/shell.cc]: `raster_task`

**Comment by liyuqian** [comment]
Thanks! I'll merge this as is to avoid conflict, and address Chinmay's comment in another PR.

**Pull request merged**